src/Filters/Filters.php

```diff
@@ -43,4 +43,14 @@ public function getIterator(): iterable
     {
         yield from $this->filters;
     }
+
+    public function getPreparedValues(ApplicatorInterface $applicator): array
```

Comment:
This has lower cognitive complexity (and better readability) and is much straightforward code:
```
public function getPreparedValues(ApplicatorInterface $applicator): array
{
    $preparedValues = [];
    foreach ($this->filters as $filter) {
        $preparedValues += $applicator->getPreparedValue($filter);
    }

    return $preparedValues;
}
```
What is benefit of this original implementation? It is much harder for me to read it.

Reply:
ok.. back to the trees.. let there is a foreach..

Reply:
:christmas_tree: :-) Thanks. Just to be clear - I'm ok with this kind of FP implementation, if it has some benefit - is much faster, is easier to read, or the "traditional" implementation would be complicated or ineffective. 

The right tool should be used for the right job :).

Reply:
Its not about me or you - the purpose of the code is not to break traditions, but to have working implementation which is readable and long-term maintainable even for newcommers and people without FP background. The "smarter" and less common (less "traditional") code, the harder to read and maintain. I don't like the fact neither, but one must live with this :-D